lib/membrane/rtp/demuxer.ex

```diff
@@ -67,25 +118,78 @@ defmodule Membrane.RTP.Demuxer do
   end
 
   @impl true
-  def handle_pad_added(Pad.ref(:output, ssrc) = pad, _ctx, state) do
-    buffered_actions = state.stream_states[ssrc].buffered_actions
-
-    state =
-      Bunch.Struct.update_in(
-        state,
-        [:stream_states, ssrc],
-        &%{&1 | phase: :linked, buffered_actions: []}
+  def handle_pad_added(Pad.ref(:output, _ref) = pad, ctx, state) do
+    matching_stream_ssrc =
+      find_matching_stream_for_pad(
+        ctx.pad_options.stream_id,
+        state.stream_states,
+        state.payload_type_mapping
       )
 
-    {[stream_format: {pad, %RTP{}}] ++ Enum.reverse(buffered_actions), state}
+    stream_state = state.stream_states[matching_stream_ssrc]
+
+    case stream_state do
+      nil ->
+        state = %{
+          state
+          | pads_waiting_for_stream:
+              Map.put(state.pads_waiting_for_stream, pad, ctx.pad_options.stream_id)
+        }
+
+        {[], state}
+
+      %{phase: :waiting_for_link} ->
+        Process.cancel_timer(stream_state.link_timer)
+
+        buffer_action = [buffer: {pad, Enum.reverse(stream_state.queued_buffers)}]
+
+        end_of_stream_action =
+          if stream_state.end_of_stream_buffered, do: [end_of_stream: pad], else: []
+
+        state =
+          Bunch.Struct.put_in(
```

Comment:
Generally use `Kernel.put_in` instead of `Bunch.Struct.put_in`

Reply:
I do, but in cases, where keys are not known in compile time, I find Bunch.Struct functions handy

Reply:
I think I don't see such cases in this PR 🤔